docs/Targets.md

```diff
@@ -0,0 +1,123 @@
+# Targets
+
+## SiFive HiFive1 RISC-V
+
+### Features
+* E31 RISC-V 320MHz 32-bit processor
+* Onboard 16KB scratchpad RAM
+* External 4MB QSPI Flash 
+
+### Default Linker Settings
+* FLASH: Address 0x20000000, Len 0x6a120 (424 KB)
+* RAM:   Address 0x80000000, Len 0x4000  (16 KB)
+
+### Stock bootloader
+Start Address: 0x20000000 is 64KB. Provides a "double tap" reset feature to halt boot and allow debugger to attach for reprogramming. Press reset button, when green light comes on press reset button again, then board will flash red.
+
+### Application Code
+Start Address: 0x20010000
+
+### wolfBoot configuration
+
+The default wolfBoot configuration will add a second stage bootloader, leaving the stock "double tap" bootloader for safety.
+
+For testing wolfBoot here are the changes required:
+
+1. Makefile arguments:
+    * ARCH=RISCV
+    * TARGET=hifive1
+ 
+    ```
+    make ARCH=RISCV TARGET=hifive1 clean
+    make ARCH=RISCV TARGET=hifive1
+    ```
+
+    If using the `riscv64-unknown-elf-` cross compiler you can add `CROSS_COMPILE=riscv64-unknown-elf-` to your `make` or modify `arch.mk` as follows:
+    
+    ```
+     ifeq ($(ARCH),RISCV)
+    -  CROSS_COMPILE:=riscv32-unknown-elf-
+    +  CROSS_COMPILE:=riscv64-unknown-elf-
+    ```
+
+
+2. `include/target.h`
+
+Bootloader Size: 0x10000 (64KB)
+
+```c
+#define WOLFBOOT_SECTOR_SIZE                 0x10000
+#define WOLFBOOT_PARTITION_BOOT_ADDRESS      0x20010000
+```
+
+Application Size 0x40000 (256KB)
+
+```c
+#define WOLFBOOT_PARTITION_SIZE              0x40000
+#define WOLFBOOT_PARTITION_UPDATE_ADDRESS    0x20020000
+#define WOLFBOOT_PARTITION_SWAP_ADDRESS      0x20060000
+```
+
+### Build Options
+
+* To use ECC instead of ED25519 use make argument `SIGN=ECC256`
+* To output wolfboot as hex for loading with JLink use make argument `wolfboot.hex`
+
+### Loading
+
+Loading with JLink:
+
+```
+JLinkExe -device FE310 -if JTAG -speed 4000 -jtagconf -1,-1 -autoconnect 1
+loadbin factory.bin 0x20010000
+rnh
+```
+
+### Debugging
+
+Debugging with JLink:
+
+In one terminal:
+`JLinkGDBServer -device FE310 -port 3333`
+
+In another terminal:
+`riscv64-unknown-elf-gdb test-app/image.elf -ex "set remotetimeout 240" -ex "target extended-remote localhost:3333"`
+or
+`riscv64-unknown-elf-gdb wolfboot.elf -ex "set remotetimeout 240" -ex "target extended-remote localhost:3333"`
+
+
+
+## STM32-F407
```

Comment:
This should be the first target described in Targets.md since it's the default (CI)